operator/duties/base_handler.go

```diff
@@ -0,0 +1,86 @@
+package duties
+
+import (
+	"context"
+
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	spectypes "github.com/bloxapp/ssv-spec/types"
+	"go.uber.org/zap"
+
+	"github.com/bloxapp/ssv/networkconfig"
+	beaconprotocol "github.com/bloxapp/ssv/protocol/v2/blockchain/beacon"
+)
+
+// ExecuteDutiesFunc is a non-blocking functions which executes the given duties.
+type ExecuteDutiesFunc func(logger *zap.Logger, duties []*spectypes.Duty)
+
+type dutyHandler interface {
+	Setup(beaconprotocol.BeaconNode, networkconfig.NetworkConfig, ValidatorController, ExecuteDutiesFunc, chan phase0.Slot, chan ReorgEvent, chan bool)
+	HandleDuties(context.Context, *zap.Logger)
+	Name() string
+	IndicesChangeChannel() chan bool
+	ReorgChannel() chan ReorgEvent
+}
+
+type baseHandler struct {
+	beaconNode          beaconprotocol.BeaconNode
+	network             networkconfig.NetworkConfig
+	validatorController ValidatorController
+	executeDuties       ExecuteDutiesFunc
+	ticker              chan phase0.Slot
+
+	reorg         chan ReorgEvent
+	indicesChange chan bool
+
+	fetchFirst     bool
```

Comment:
Should this be atomic?

Reply:
no, the code designed to not write in parallel